packages/engine/src/avatar/AvatarTeleportSystem.ts

```diff
@@ -0,0 +1,124 @@
+// import { ParityValue } from 'src/common/enums/ParityValue'
+// import { getHandPosition } from 'src/xr/functions/WebXRFunctions'
+import {
+  AdditiveBlending,
+  BufferAttribute,
+  BufferGeometry,
+  Group,
+  Line,
+  LineBasicMaterial,
+  PointLight,
+  Vector3
+} from 'three'
+
+import { normalizeRange } from '@xrengine/common/src/utils/mathUtils'
+
+import { Engine } from '../ecs/classes/Engine'
+import { World } from '../ecs/classes/World'
+import { defineQuery, getComponent } from '../ecs/functions/ComponentFunctions'
+import { XRInputSourceComponent } from '../xr/components/XRInputSourceComponent'
+import { AvatarTeleportTagComponent } from './components/AvatarTeleportTagComponent'
+import { teleportAvatar } from './functions/moveAvatar'
+
+// Guideline parabola function
+const positionAtT = (inVec: Vector3, t: number, p: Vector3, v: Vector3, gravity: Vector3): Vector3 => {
+  inVec.copy(p)
+  inVec.addScaledVector(v, t)
+  inVec.addScaledVector(gravity, 0.5 * t ** 2)
+  return inVec
+}
+
+// Utility Vectors
+// Unit: m/s
+const initialVelocity = 6
+const dynamicVelocity = 10
+const gravity = new Vector3(0, -9.8, 0)
+const tempVec = new Vector3()
+const tempVec1 = new Vector3()
+const tempVecP = new Vector3()
+const tempVecV = new Vector3()
+
+const getParabolaInputParams = (
+  controller: Group,
+  initialVelocity: number,
+  gravity: Vector3
+): { p: Vector3; v: Vector3; t: number } => {
+  // Controller start position
+  const p = controller.getWorldPosition(tempVecP)
+
+  // Set Vector V to the direction of the controller, at 1m/s
+  const v = controller.getWorldDirection(tempVecV)
+
+  // Scale the initial velocity
+  let normalizedYDirection = 1 - normalizeRange(v.y, -1, 1)
+  v.multiplyScalar(initialVelocity + dynamicVelocity * normalizedYDirection)
+
+  // Time for tele ball to hit ground
+  const t = (-v.y + Math.sqrt(v.y ** 2 - 2 * p.y * gravity.y)) / gravity.y
```

Comment:
Probably a good idea to check for division by zero.

Reply:
gravity is a fixed vector. y will always be -9.8 so that's why